cupy/core/core.pyx

```diff
@@ -1921,6 +1956,43 @@ cpdef ndarray _take(ndarray a, indices, axis=None, ndarray out=None):
             a.reduced_view(), indices, cdim, rdim, adim, index_range, out)
 
 
+cpdef _put(ndarray a, ind, v, axis=None):
```

Comment:
I am planning to use `_put` with `axis` when implementing `__setitem__` with array indexing in the future.